packages/cli/generation/ir-generator/src/formatDocs.ts

```diff
@@ -1,13 +1,16 @@
 const isBrowser = typeof window !== "undefined" && typeof window.document !== "undefined";
 
-export async function formatDocs(docs: string | undefined): Promise<string | undefined> {
+export function formatDocs(docs: string | undefined): string | undefined {
     if (docs == null || isBrowser) {
         return undefined;
     }
-    const prettier = await import("prettier");
-    const formattedDocs = prettier.format(docs, {
-        parser: "markdown"
-    });
+    // TODO: Find a way to support this in the browser without an async conditional import.
+    //
+    // const prettier = import("prettier");
+    // const formattedDocs = prettier.format(docs, {
+    //     parser: "markdown"
+    // });
```

Comment:
can you have the caller pass in a function to `generateIntermediateRepresentation` here ?

Reply:
then we can only pass it in for sdk generation but leave it out for snippets

Reply:
I took a look at that earlier, but it still requires calling an `async` function (even when it's abstracted away). I experimented with moving the docs formatting up a layer - we can handle this in the workspace before its passed onto the IR generator so that the IR generator can remain totally sync.

I explored doing this in the OSSWorkspace and LazyFernWorkspace [here](https://github.com/fern-api/fern/pull/5363/commits/b318ba849dae417a1a1c18b3574c247d116aca29), but if this looks good I think we could  move it to the `AbstractAPIWorkspace` as a protected method so that every `async` variant has access to the same implementation in a single place.

Let me know what you think!

Reply:
Resolved in [this comment](https://github.com/fern-api/fern/pull/5363#discussion_r1876188319).